app/components/dashboard/compose/left-panel/component.js

```diff
@@ -35,23 +59,116 @@ export default Component.extend({
       return;
     }
     events.on('select', function (allSelected) {
+      allSelected = self.insertPackageName(allSelected);
       self.set('inputData', allSelected);
     });
     events.on('selectEnvironment', function (selectedEnvironment) {
-      self.set('selectedEnvironment', selectedEnvironment);
+      if (self) {
+        self.set('selectedEnvironment', selectedEnvironment);
+      }
     });
+
+    this.setDefaults();
+    // Check if we're ingesting a dataset/tale
+    if (this.get('model').queryParams.data_location) {
+      this.set('importing', true);
+      this.setThirdPartyParams();
+      this.renderImportInfo();
+    }
   },
 
-  // just checking the toggle works ...
-  publicCheckedObserver: observer('public_checked', function () {
-    console.log("Checked = " + this.get('public_checked'));
-  }),
+  /**
+   * Resets the GUI.
+   * @method getFinalJobStatus
+   */
+  setDefaults() {
+    // Set default values of mutable variables
+    this.set('progress', 0);
+    this.set('launchingInstance', false);
+    this.set('message', 'Launching new Tale, please wait.');
+    this.set('errorMessage', '');
+    this.set('datasetTitle', '');
+  },
 
-  publicDescriptionObserver: observer('description', function () {
-    console.log("Description = " + this.get('description'));
-  }),
+  renderImportInfo() {
+      /* 
+        Schedules the function that fills in the Tale name to run afterRender
+      */
+    
+    scheduleOnce('afterRender', this, () => {
+      // Fill in the dataset title in the Tale Name and Input data fields
+      this.renderExternalPackage();
+      // Let the right panel component know that it should disable the file browser
+      this.get('wtEvents').events.disableRightPanel();
+      if (this.get('model').queryParams && this.get('model').queryParams.environment) {
+        this.get('wtEvents').events.selectEnvironmentByName(this.get('model').queryParams.environment);
+      }
+      });
+  },
 
-  // public_checked : false,
+  setThirdPartyParams() {
+    // If the user is coming to this page from a third party with the intent
+    // on importing data, the referrer is set here.
+    this.set('datasetLocation', this.get('model').queryParams.data_location);
+    this.set('datasetTitle', this.get('model').queryParams.data_title);
+    this.set('packageAPI', this.get('model').queryParams.data_api);
```

Comment:
I've since added checks to see which ones are appropriate to set; I also passed `packageAPI` to the backend as a kwarg to handle the case where we want to define an endpoint that should be used to find the package. 

I'll update readthedocs to reflect this change.